server/protocols/webadmin/webadmin-jmap/src/test/java/org/apache/james/webadmin/data/jmap/RecomputeMessageFastViewProjectionItemsRequestToTaskTest.java

```diff
@@ -0,0 +1,322 @@
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+import spark.Service;
+
+class RecomputeMessageFastViewProjectionItemsRequestToTaskTest {
+    private final class JMAPRoutes implements Routes {
+        private final MessageFastViewProjectionCorrector corrector;
+        private final TaskManager taskManager;
+        private final MessageId.Factory factory;
+
+        private JMAPRoutes(MessageFastViewProjectionCorrector corrector, TaskManager taskManager, MessageId.Factory factory) {
+            this.corrector = corrector;
+            this.taskManager = taskManager;
+            this.factory = factory;
+        }
+
+        @Override
+        public String getBasePath() {
+            return BASE_PATH;
+        }
+
+        @Override
+        public void define(Service service) {
+            service.post(BASE_PATH,
+                TaskFromRequestRegistry.builder()
+                    .registrations(new RecomputeMessageFastViewProjectionItemsRequestToTask(corrector, factory))
+                    .buildAsRoute(taskManager),
+                new JsonTransformer());
+        }
+    }
+
+    static final String BASE_PATH = "/messages/:messageId";
+
+    static final MessageFastViewPrecomputedProperties PROJECTION_ITEM = MessageFastViewPrecomputedProperties.builder()
+        .preview(Preview.from("body"))
+        .hasAttachment(false)
+        .build();
+
+    static final DomainList NO_DOMAIN_LIST = null;
+    static final Username BOB = Username.of("bob");
+    static final Username CEDRIC = Username.of("cedric");
+    static final String MESSAGE_ID = "1";
+
+    WebAdminServer webAdminServer;
+    MemoryTaskManager taskManager;
+    MemoryMessageFastViewProjection messageFastViewProjection;
+    InMemoryMailboxManager mailboxManager;
+    MemoryUsersRepository usersRepository;
+    MessageId.Factory factory;
+
+    @BeforeEach
+    void setUp() {
+        factory = new InMemoryMessageId.Factory();
+        JsonTransformer jsonTransformer = new JsonTransformer();
+        taskManager = new MemoryTaskManager(new Hostname("foo"));
+
+        messageFastViewProjection = new MemoryMessageFastViewProjection(new RecordingMetricFactory());
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        usersRepository = MemoryUsersRepository.withoutVirtualHosting(NO_DOMAIN_LIST);
+        MessageContentExtractor messageContentExtractor = new MessageContentExtractor();
+        HtmlTextExtractor htmlTextExtractor = new JsoupHtmlTextExtractor();
+        Preview.Factory previewFactory = new Preview.Factory(messageContentExtractor, htmlTextExtractor);
+        MessageFastViewPrecomputedProperties.Factory projectionItemFactory = new MessageFastViewPrecomputedProperties.Factory(previewFactory);
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new TasksRoutes(taskManager, jsonTransformer),
+            new JMAPRoutes(
```

Comment:
usually we test the real routes not some test ones. Why do you need to do that here?

Reply:
That's something bad we have been carrying from other preview recomputation tasks as well... Will fix, thanks for pointing it out